src/stcal/dark_current/dark_sub.py

```diff
@@ -0,0 +1,391 @@
+#
+#  Module for dark subtracting science data sets
+#
+
+import copy
+import numpy as np
+import logging
+
+from . import dark_class
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def do_correction(input_model, dark_model, dark_output=None):
+    """
+    Convert models to classes to remove internal dependence on data models.
+    After the creation of internal classes, the dark current subtraction is
+    done.
+
+    Parameters
+    ----------
+    input_model : data model
+        Input model assumed to be like a JWST RampModel.
+
+    dark_model : data model
+        Input model assumed to be like a JWST DarkModel.
+
+    dark_output : str
+        A path to denote where to save the dark model, if desired.
+
+    Returns
+    -------
+    output_data : ScienceData
+        dark-subtracted science data
+
+    averaged_dark : DarkData
+        New dark object with averaged frames
+    """
+    science_data = dark_class.ScienceData(input_model)
+    dark_data = dark_class.DarkData(dark_model=dark_model)
+
+    return do_correction_data(science_data, dark_data, dark_output)
+
+
+def do_correction_data(science_data, dark_data, dark_output=None):
+    """
+    Short Summary
+    -------------
+    Execute all tasks for Dark Current Subtraction
+
+    Parameters
+    ----------
+    science_data : ScienceData
+        science data to be corrected
+
+    dark_data : dark model object
+        dark data
+
+    dark_output : string
+        file name in which to optionally save averaged dark data
+
+    Returns
+    -------
+    output_data : ScienceData
+        dark-subtracted science data
+
+    averaged_dark : DarkData
+        New dark object with averaged frames
+    """
+
+    # Save some data params for easy use later
+    instrument = science_data.instrument_name
+    sci_nints = science_data.data.shape[0]
+    sci_ngroups = science_data.data.shape[1]
+    sci_nframes = science_data.exp_nframes
+    sci_groupgap = science_data.exp_groupgap
+
+    if instrument == 'MIRI':
```

Comment:
It might be possible to remove the instrument-dependent code here by looking at len(dark_data.data.shape). If it's 3, then it's not MIRI data and drk_nints=1, drk_ngroups=shape[0]. If it's 4, then drk_nints=shape[0], drk_ngroups=shape[1].